### PR TITLE
common/LogEntry: replace ceph_entity_inst_t with name + addrvec

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
@jcsp I'm wondering if we should remove the IP address from these log messages.  (It's about to turn into a list of addresses, too, if the daemon is ipv4+ipv6 multi-homed *or* include an address twice for the msgr1->msgr2 transition.)

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2018-05-26_16:13:44-rados-wip-sage2-testing-2018-05-26-0833-distro-basic-smithi/

File: src/msg/msg_types.h [hunk 1]
```diff
@@ -551,6 +551,27 @@ struct entity_addrvec_t {
...
   }
+
+  friend bool operator==(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    return l.v == r.v;
+  }
+  friend bool operator!=(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    return l.v != r.v;
+  }
+  friend bool operator<(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    for (unsigned p = 0; l.v.size() >= p && r.v.size() >= p; ++p) {
```
Comment (tchaikov) [inline src/msg/msg_types.h]: could just use the `vector` 's overloaded operator defined in `<vector>`, it performs `lexicographical_compare()` under the hood. it's equivalent to what we have here.

File: src/common/LogEntry.h [hunk 2]
```diff
@@ -173,8 +171,8 @@ inline ostream& operator<<(ostream& out, const clog_type t)
 
 inline ostream& operator<<(ostream& out, const LogEntry& e)
 {
-  return out << e.stamp << " " << e.name << " " << e.who
-	     << " " << e.seq << " : "
+  return out << e.stamp << " " << e.name << " (" << e.rank
+	     << " " << e.addrs << ") " << e.seq << " : "
```
Comment (tchaikov) [inline src/common/LogEntry.h]: do we really want to print out the `addrs` in each log message? i think this slows down the logging even more.

File: src/test/test_addrs.cc [hunk 3]
```diff
@@ -237,3 +238,30 @@ TEST(Msgr, TestAddrvecEncodeAddrDecode3)
...
+
+TEST(entity_addrvec_t, parse)
+{
+  entity_addrvec_t addrvec;
+  bufferlist bl;
+
+  for (auto v : { addr_checks, addr_checks2, addrvec_parse_checks }) {
+    for (unsigned i = 0; v[i][0]; ++i) {
+      const char *end = "";
+      addrvec.parse(v[i][0], &end);
```
Comment (tchaikov) [inline src/test/test_addrs.cc]: could 
```c
ASSERT_TRUE(addrvec.parse(v[i][0], &end));
```
here

File: src/test/test_addrs.cc [hunk 4]
```diff
@@ -237,3 +238,30 @@ TEST(Msgr, TestAddrvecEncodeAddrDecode3)
...
+  { "127.0.0.1 foo", "127.0.0.1:0/0", " foo" },
+  { "127.0.0.1 1.2.3.4 foo", "[127.0.0.1:0/0,1.2.3.4:0/0]", " foo" },
+  { "127.0.0.1 :: - foo", "[127.0.0.1:0/0,[::]:0/0,-]", " foo" },
+  { NULL, NULL, NULL },
+};
+
+TEST(entity_addrvec_t, parse)
+{
+  entity_addrvec_t addrvec;
+  bufferlist bl;
```
Comment (tchaikov) [inline src/test/test_addrs.cc]: nit, `bl` is not used.

File: src/test/test_addrs.cc (hunk 3 above)
Comment (tchaikov) [reply src/test/test_addrs.cc]: but all of them contain one or more addresses in it. in other words, none of the test cases just contains invalid address.

**Comment by tchaikov** [comment]
retest this please.

File: src/test/test_addrs.cc (hunk 3 above)
Comment (liewegas) [reply src/test/test_addrs.cc]: the 1.2.3 case didn't.  i added a return value check though that was also conditional on out.empty()

**Pull request merged**